File: Todo-Main-Objc-C/Todo-Main-Objc-C/ViewController.m
//
//  ViewController.m
//  Todo-Main-Objc-C
//
//  Created by topik mujianto on 07/02/20.
//  Copyright © 2020 topik mujianto. All rights reserved.
//

#import "ViewController.h"
@import GITSTodo;

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UITableView *tableView;
@property (nonatomic) NSArray<NSString *> *todos;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [NetworkManager getTodosOnSuccess:^(NSArray<NSString *> *result){
        self.todos = result;
        [self.tableView reloadData];
    } onError:^(NSString *error){
        UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Error"
                                       message: error
                                       preferredStyle:UIAlertControllerStyleAlert];

        UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
           handler:^(UIAlertAction * action) {}];

        [alert addAction:defaultAction];
        [self presentViewController:alert animated:YES completion:nil];
    }];
}

//MARK- UITableViewDataSource
- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"CELL_TODO"];
    if (cell == nil) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"CELL_TODO"];
    }
    cell.textLabel.text = self.todos[indexPath.row];
    return cell;
}

- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.todos.count;
}


@end

File: Todo-Main-Swift/Todo-Main-Swift/ViewController.swift
//
//  ViewController.swift
//  Todo-Main-Swift
//
//  Created by topik mujianto on 07/02/20.
//  Copyright © 2020 topik mujianto. All rights reserved.
//

import UIKit
import GITSTodo

class ViewController: UIViewController {

    @IBOutlet weak var tableView: UITableView!
    fileprivate var todos = [String]()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        NetworkManager.getTodos(onSuccess: { [weak self] result in
            guard let `self` = self else {return}
            self.todos = result
            self.tableView.reloadData()
        }, onError: { [weak self] error in
            guard let `self` = self else {return}
            let alert = UIAlertController(title: "Error", message: error, preferredStyle: .alert)
            alert.addAction(UIAlertAction(title: "OK", style: .cancel, handler: nil))
            self.present(alert, animated: true, completion: nil)
        })
        
    }
}

//MARK:- UITableViewDataSource
extension ViewController: UITableViewDataSource {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.todos.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "CELL_TODO")
        let todo = self.todos[indexPath.row]
        cell?.textLabel?.text = todo
        return cell ?? UITableViewCell()
    }
    
    
}

